src/juju/jimm-facade.ts

```diff
@@ -3,10 +3,45 @@ import { autoBind } from "@canonical/jujulib/dist/api/utils";
 
 import type { Controller } from "store/juju/types";
 
+// As typed in JIMM:
+// https://github.com/canonical/jimm/blob/1da76ed2d8dba741f5880f32c073f85f7d518904/api/params/params.go#L92
+export type AuditEvent<P = unknown, E = unknown> = {
+  "conversation-id": string;
+  errors: Record<string, E> | null;
+  "facade-method": string;
+  "facade-name": string;
+  "facade-version": number;
+  "is-response": boolean;
+  "message-id": number;
+  model: string;
+  "object-id": string;
+  params: Record<string, P> | null;
+  time: string;
+  "user-tag": string;
+};
+
+// As typed in JIMM:
+// https://github.com/canonical/jimm/blob/1da76ed2d8dba741f5880f32c073f85f7d518904/api/params/params.go#L132
+export type AuditEvents = {
+  events: AuditEvent[];
```

Comment:
I'm going to land this PR without that change, but if you manage to get a null response let me know and I'll fix it up (I've only ever received an array and if I'm reading the type correctly it should always return something but it could be wrong).